test/Mono.Linker.Tests.Cases/RequiresCapability/RequiresCapability.cs

```diff
@@ -1228,6 +1228,8 @@ class RequiresOnMethod
 			}
 
 			[ExpectedWarning ("IL2109", "RequiresOnClass/DerivedWithoutRequires", "RequiresOnClass.ClassWithRequires", "--ClassWithRequires--", ProducedBy = ProducedBy.Trimmer)]
+			[ExpectedWarning ("IL2109", "RequiresOnClass.DerivedWithoutRequires", "RequiresOnClass.ClassWithRequires", "--ClassWithRequires--", ProducedBy = ProducedBy.Analyzer)]
```

Comment:
If the warning is produced by both the Trimmer and the Analyzer you can get rid of specifying the ProducedBy which by default is ProducedBy.TrimmerAndAnalyzer
```suggestion
			[ExpectedWarning ("IL2109", "RequiresOnClass.DerivedWithoutRequires", "RequiresOnClass.ClassWithRequires", "--ClassWithRequires--")]
```

Reply:
Also as a nit some of the changes add an additional new line

Reply:
I just read the comment on why they are duplicated, seems to me like we should open something on linker to track this discrepancy, it doesn't make a lot of sense that we first print "/" and then "." for the same message.

Reply:
Made a new issue for it: https://github.com/dotnet/linker/issues/2441

It looks like FullName and GetDisplayName() are the source of the discrepancy